libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -495,6 +495,25 @@ void RC_Channel::init_aux_function(const aux_func_t ch_option, const aux_switch_
     }
 }
 
+#if !HAL_MINIMIZE_FEATURES
+
+/* lookup the announcement for switch change*/
+const char *RC_Channel::string_for_aux_function(AUX_FUNC function) const     
+{
+    for (int k=0; 48; k++) {     //wish I knew how to determine the # rows in the table!
+        if (lookuptable[k].option == AUX_FUNC::DO_NOTHING) {
+            return "UNKNOWN";
+        } else {
+            if (lookuptable[k].option == function) {
+            return lookuptable[k].announcement;
+            }
+         }
+        return "UNKNOWN";
+    }
```

Comment:
```suggestion
    for (struct LookupTable entry : lookuptable) {
        if (entry.option == function) {
            return entry.announcement;
        }
    }
    return "UNKNOWN";
```

This should not need any sentinels, and actually will compile to be the same code as a for loop with an index. Then all you have to do is update the list that you are scanning through in the header when adding to this.

Reply:
super thanks!